Crypto

thvdveld opened the pull request:
This adds support for the crypto accelerator available:

- [ ] AES:
  - [ ] ECB
  - [ ] CBC
  - [X] CTR
  - [ ] GCM
  - [X] CCM
  - [ ] CBC-MAC
- [x] SHA-256
- [x] ECC:
  - [x] ECC-MUL
  - [x] ECC-ADD
- [ ] Big Number operations:
  - [x] Add
  - [x] Subtract
  - [ ] AddSubtract
  - [x] Multiply
  - [ ] Divide
  - [ ] Modulo
  - [x] Compare
  - [ ] Copy
  - [ ] Left shift
  - [ ] Right Shift
  - [ ] Inverse Modulo:
    The inverse modulo operation requires the modulus to be odd, otherwise the crypto accelerator will not work as exepected.
    - [ ] With even modulus
    - [ ] With odd modulus
    - [ ] With prime modulus

thvdveld committed:
start on ecc-mul

thvdveld committed:
elliptic curve addition and multiplication

thvdveld committed:
working aes example

thvdveld committed:
restructure code

thvdveld committed:
aes-ccm: add decrypt function

thvdveld committed:
api changes

thvdveld committed:
make crypto accelerator non generic

thvdveld committed:
add modular exponent calculation

thvdveld committed:
make allocation a feature

thvdveld committed:
update bignum operations

thvdveld committed:
Add read to UART

thvdveld committed:
Removed: stabelized feature

thvdveld committed:
Added: derive Eq when already deriving PartialEq

thvdveld committed:
Cargo fmt

thvdveld committed:
Start with BigNum API

thvdveld committed:
Rewrite inner workings

chrysn commented:
Even if this is incomplete in terms of supported operations, it appears that it has supported the [edhoc-rs](https://github.com/openwsn-berkeley/edhoc-rs) project well so far.

What is missing for this to be merged and a version of the HAL with crypto support to be published?

thvdveld commented:
It's been some time since I last looked at this, so I have no idea. I'll try to take a look at it and push a new version with crypto support.

thvdveld committed:
fix: add ConstParamTy to enum

This is required to be able to use that type as a const generic.
Since there is not much activity on that feature in the Rust compiler,
I'll try to remove this feature in this crate.

thvdveld committed:
rebase on master
